### PR TITLE
FIX: strip the trailing slash (/) of cors origins.

### PR DESCRIPTION
**Pull request by vinothkannans**
And also a validation is added for site setting.

The CORS functionality is not working as expected if the trailing slash is not stripped.

### CODE REVIEW

File: spec/components/hijack_spec.rb [hunk 1]
```diff
@@ -116,6 +116,42 @@ def hijack_test(&blk)
     expect(headers).to eq(expected)
   end
 
+  it "removes trialing slash in cors origin" do
```
Comment (jjaffeux) [inline spec/components/hijack_spec.rb:119]:
```suggestion
  it "removes trailing slash in cors origin" do
```

**Review by eviltrout — APPROVED** [review]
Looks good!

File: config/initializers/008-rack-cors.rb [hunk 2]
```diff
@@ -34,6 +34,8 @@ def self.apply_headers(cors_origins, env, headers)
     origin = nil
 
     if cors_origins
+      cors_origins = cors_origins.map { |x| x.chomp('/') }
+
```
Comment (SamSaffron) [inline config/initializers/008-rack-cors.rb:38]: should we not chomp it a bit earlier at ` cors_origins += SiteSetting.cors_origins.split('|') if SiteSetting.cors_origins.present?`

Comment (SamSaffron) [reply config/initializers/008-rack-cors.rb:38]: I guess if we keep the global origins pre-chomped ... maybe all of this extra work can be just totally avoided by adding a validation on the site setting?

Comment (vinothkannans) [reply config/initializers/008-rack-cors.rb]: The site setting validation is now added.

**Review by SamSaffron — APPROVED** [review]

**Comment by SamSaffron** [comment]
Looks perfect to me!

**Pull request merged**